src/Symfony/Component/HttpKernel/EventListener/ValidationFailedExceptionListener.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\ExceptionEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Serializer\Exception\NotEncodableValueException;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\Validator\Exception\ValidationFailedException;
+
+/**
+ * @author Konstantin Myakshin <molodchick@gmail.com>
+ */
+final class ValidationFailedExceptionListener implements EventSubscriberInterface
+{
+    public function __construct(private readonly SerializerInterface $serializer)
+    {
+    }
+
+    public function onKernelException(ExceptionEvent $event): void
+    {
+        $throwable = $event->getThrowable();
+        if (!$throwable instanceof ValidationFailedException) {
+            return;
+        }
+
+        $request = $event->getRequest();
+        $format = $request->getPreferredFormat('json');
+
+        try {
+            $data = $this->serializer->serialize($throwable->getViolations(), $format);
+        } catch (NotEncodableValueException) {
+            return;
+        }
+
+        $response = new Response(
+            $data,
+            Response::HTTP_UNPROCESSABLE_ENTITY,
```

Comment:
it's reasonable to allow for a custom status code from eg. ValidationFailedException::getStatusCode()

Reply:
but `ValidationFailedException` throwed from `RequestPayloadValueResolver`, you can't throw your own special exception from it.